subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -62,7 +66,8 @@ public FileSystemLocationSnapshot snapshot(String absolutePath, @Nullable Snapsh
         final MerkleDirectorySnapshotBuilder builder = MerkleDirectorySnapshotBuilder.sortingRequired();
```

Comment:
I wonder if we should move the builder to the visitor as well, and then have a `getResult()` method in the `visitor`.

Reply:
Great idea, done